### ISSUE TITLE
master fails to build from source

### ISSUE THREAD

**Issue by vedranmiletic** [ugeneunipro/ugene#47]
Fedora 30, GCC 9.1.1, cmake 3.14.5, GNU Make 4.2.1, ninja 1.9.0.

```
[623/2312] Building CXX object src/corelibs/U2Formats/CMakeFiles/U2Formats.dir/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.cpp.o
FAILED: src/corelibs/U2Formats/CMakeFiles/U2Formats.dir/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.cpp.o 
/usr/lib64/ccache/c++  -DBUILDING_U2FORMATS_DLL -DQT_CORE_LIB -DQT_FATAL_ASSERT -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_SCRIPT_LIB -DQT_SQL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DU2Formats_EXPORTS -DUGENE_USE_BUNDLED_SQLITE -DUGENE_USE_BUNDLED_ZLIB -DUGENE_VERSION=33.0-dev -DUGENE_VER_MAJOR=33 -DUGENE_VER_MINOR=0 -DUGENE_WEB_KIT -Isrc/corelibs/U2Formats/U2Formats_autogen/include -I../src/corelibs/U2Formats/src -I../src/corelibs/U2Formats/../../include -I../src/corelibs/U2Formats/../../libs_3rdparty/samtools/src -I../src/corelibs/U2Formats/../../libs_3rdparty/samtools/src/samtools -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtSql -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtScript -isystem /usr/include/qt5/QtXml -fPIC   -fPIC -std=gnu++11 -MD -MT src/corelibs/U2Formats/CMakeFiles/U2Formats.dir/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.cpp.o -MF src/corelibs/U2Formats/CMakeFiles/U2Formats.dir/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.cpp.o.d -o src/corelibs/U2Formats/CMakeFiles/U2Formats.dir/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.cpp.o -c ../src/corelibs/U2Formats/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.cpp
In file included from ../src/corelibs/U2Formats/../../include/U2Core/../../corelibs/U2Core/src/globals/U2SafePoints.h:34,
                 from ../src/corelibs/U2Formats/../../include/U2Core/U2SafePoints.h:1,
                 from ../src/corelibs/U2Formats/../../include/U2Core/../../corelibs/U2Core/src/dbi/U2AbstractDbi.h:39,
                 from ../src/corelibs/U2Formats/../../include/U2Core/U2AbstractDbi.h:1,
                 from ../src/corelibs/U2Formats/src/sqlite_dbi/assembly/../SQLiteDbi.h:27,
                 from ../src/corelibs/U2Formats/src/sqlite_dbi/assembly/../SQLiteAssemblyDbi.h:27,
                 from ../src/corelibs/U2Formats/src/sqlite_dbi/assembly/SingleTableAssemblyAdapter.h:25,
                 from ../src/corelibs/U2Formats/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.h:25,
                 from ../src/corelibs/U2Formats/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.cpp:22:
../src/corelibs/U2Formats/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.cpp: In member function „void U2::MultiTablePackAlgorithmAdapter::migrate(U2::MTASingleTableAdapter*, const QVector<U2::SQLiteReadTableMigrationData>&, qint64, qint64, U2::U2OpStatus&)”:
../src/corelibs/U2Formats/src/sqlite_dbi/assembly/MultiTableAssemblyAdapter.cpp:682:24: greška: „newProwRegion” was not declared in this scope
  682 |                 assert(newProwRegion.contains(d.newProw));
      |                        ^~~~~~~~~~~~~
[655/2312] Building CXX object src/corelibs/U2Lang/CMakeFiles/U2Lang.dir/src/support/serialize/HRSchemaSerializer.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by DmitriiSukhomlinov** [comment]
Hello!
I made a fast fix, make sure that it works. If it's not - please, feel free to contact us.
One more thing - cmake isn't good tested for the build process. I advice you to use qmake - it's more reliable and definitely works well.
Also, can you, please, say - are you trying to build debug or release?

Best regards,
UGENE team

**Comment by vedranmiletic** [comment]
I'm building whatever is the default setting, I don't specify anything. I would prefer to use CMake as many software packages use it and I'm more familiar with it than with qmake.

So, this is now fixed, thanks! Now I get the following linker error:

```
[1538/2315] Linking CXX executable src/ugeneui/ugeneui
FAILED: src/ugeneui/ugeneui 
: && /usr/lib64/ccache/c++     src/ugeneui/CMakeFiles/ugeneui.dir/ugeneui_autogen/mocs_compilation.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/Main.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/app_settings/AppSettingsDialogController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/app_settings/AppSettingsGUIImpl.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/app_settings/directories_settings/DirectoriesSettingsGUIController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/app_settings/format_settings/FormatSettingsGUIController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/app_settings/logview_settings/LogSettingsGUIController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/app_settings/network_settings/NetworkSettingsGUIController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/app_settings/resource_settings/ResourceSettingsGUIController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/app_settings/user_apps_settings/UserApplicationsSettingsGUIController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/AboutDialogController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/CheckUpdatesTask.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/DockManagerImpl.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/DockWidgetPainter.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/MDIManagerImpl.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/MainWindowImpl.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/MenuManager.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/ShutdownTask.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/SplashScreen.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/TmpDirChangeDialogController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/main_window/ToolBarManager.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/plugin_viewer/PluginViewerController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/plugin_viewer/PluginViewerImpl.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_support/DocumentFormatSelectorController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_support/DocumentProviderSelectorController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_support/DocumentReadingModeSelectorController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_support/ExportProjectDialogController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_support/MultipleDocumentsReadingModeSelectorController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_support/ProjectImpl.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_support/ProjectLoaderImpl.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_support/ProjectServiceImpl.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_support/ProjectTasksGui.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_view/BuiltInObjectViews.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/project_view/ProjectViewImpl.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/shtirlitz/Shtirlitz.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/shtirlitz/StatisticalReportController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/task_view/TaskStatusBar.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/task_view/TaskViewController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/update/UgeneUpdater.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/welcome_page/WelcomePageJsAgent.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/welcome_page/WelcomePageMdi.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/welcome_page/WelcomePageMdiController.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/src/welcome_page/WelcomePageWidget.cpp.o src/ugeneui/CMakeFiles/ugeneui.dir/qrc_ugeneui.cpp.o  -o src/ugeneui/ugeneui  -Wl,-rpath,/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2View:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Test:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Lang:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Gui:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Private:/home/vedranm/workspace/ugene/build-ninja/src/libs_3rdparty/breakpad:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Formats:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Algorithm:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Core:/home/vedranm/workspace/ugene/build-ninja/src/libs_3rdparty/sqlite3 /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Script.so.5.12.4 /usr/lib64/libQt5WebKitWidgets.so.5.212.0 src/libs_3rdparty/samtools/libsamtools.a src/corelibs/U2View/libU2View.so src/corelibs/U2Test/libU2Test.so /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5WebKit.so.5.212.0 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Network.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 src/corelibs/U2Lang/libU2Lang.so /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Script.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 src/corelibs/U2Gui/libU2Gui.so /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Svg.so.5.12.4 /usr/lib64/libQt5WebKitWidgets.so.5.212.0 /usr/lib64/libQt5PrintSupport.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5WebKit.so.5.212.0 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Network.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 src/corelibs/U2Private/libU2Private.so src/libs_3rdparty/breakpad/libbreakpad.so src/corelibs/U2Formats/libU2Formats.so src/corelibs/U2Algorithm/libU2Algorithm.so src/libs_3rdparty/samtools/libsamtools.a /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Sql.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Svg.so.5.12.4 /usr/lib64/libQt5WebKitWidgets.so.5.212.0 /usr/lib64/libQt5WebKit.so.5.212.0 /usr/lib64/libQt5Network.so.5.12.4 /usr/lib64/libQt5PrintSupport.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 src/corelibs/U2Core/libU2Core.so src/libs_3rdparty/sqlite3/libugenedb.so src/libs_3rdparty/zlib/libzlib.a /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Network.so.5.12.4 /usr/lib64/libQt5Script.so.5.12.4 /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 && :
/usr/bin/ld: src/ugeneui/CMakeFiles/ugeneui.dir/src/Main.cpp.o: in function `U2::DashboardInfoRegistry::DashboardInfoRegistry()':
Main.cpp:(.text._ZN2U221DashboardInfoRegistryC2Ev[_ZN2U221DashboardInfoRegistryC5Ev]+0x20): undefined reference to `vtable for U2::DashboardInfoRegistry'
collect2: greška: ld returned 1 exit status
[1550/2315] Linking CXX shared library src/plugins/dna_export/libdna_export.so
ninja: build stopped: subcommand failed.
```

**Comment by DmitriiSukhomlinov** [comment]
This problem appears cuz ugeneui has dependencies from U2Designer, but it wasn't shown in the CMakeList.txt file. I added it, so you can try again.
Unfortunately, possible, it's not the only place where some problems with cmake might appear. I created [a general issue](https://ugene.dev/tracker/browse/UGENE-6515) to support cmake in UGENE. But we have a release coming now, so, there is no time and possibility to make it quick. We will resolve this issue sooner or later, but, it's difficult to say when exactly. 
Also, if you inform us about each new error appears, I'm sure, we will fix them all)

Best regards,
Dmitrii Sukhomlinov,
UGENE team

**Comment by vedranmiletic** [comment]
Great progress. Thank you for opening the issue, we're very close to making it actually compile. Now it fails with:

```
[2315/2315] Linking CXX executable src/ugenecl/ugenecl
FAILED: src/ugenecl/ugenecl 
: && /usr/lib64/ccache/c++     src/ugenecl/CMakeFiles/ugenecl.dir/ugenecl_autogen/mocs_compilation.cpp.o src/ugenecl/CMakeFiles/ugenecl.dir/src/DumpHelpTask.cpp.o src/ugenecl/CMakeFiles/ugenecl.dir/src/DumpLicenseTask.cpp.o src/ugenecl/CMakeFiles/ugenecl.dir/src/DumpVersionTask.cpp.o src/ugenecl/CMakeFiles/ugenecl.dir/src/ForeverTask.cpp.o src/ugenecl/CMakeFiles/ugenecl.dir/src/Main.cpp.o src/ugenecl/CMakeFiles/ugenecl.dir/src/TaskStatusBar.cpp.o src/ugenecl/CMakeFiles/ugenecl.dir/src/TestStarter.cpp.o  -o src/ugenecl/ugenecl  -Wl,-rpath,/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2View:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Test:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Lang:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Gui:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Private:/home/vedranm/workspace/ugene/build-ninja/src/libs_3rdparty/breakpad:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Formats:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Algorithm:/home/vedranm/workspace/ugene/build-ninja/src/corelibs/U2Core:/home/vedranm/workspace/ugene/build-ninja/src/libs_3rdparty/sqlite3 /usr/lib64/libQt5Gui.so.5.12.4 src/libs_3rdparty/samtools/libsamtools.a src/corelibs/U2View/libU2View.so src/corelibs/U2Test/libU2Test.so /usr/lib64/libQt5Core.so.5.12.4 src/corelibs/U2Lang/libU2Lang.so /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Script.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 src/corelibs/U2Gui/libU2Gui.so /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Svg.so.5.12.4 /usr/lib64/libQt5WebKitWidgets.so.5.212.0 /usr/lib64/libQt5PrintSupport.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5WebKit.so.5.212.0 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Network.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 src/corelibs/U2Private/libU2Private.so src/libs_3rdparty/breakpad/libbreakpad.so src/corelibs/U2Formats/libU2Formats.so src/corelibs/U2Algorithm/libU2Algorithm.so src/libs_3rdparty/samtools/libsamtools.a /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Sql.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Svg.so.5.12.4 /usr/lib64/libQt5WebKitWidgets.so.5.212.0 /usr/lib64/libQt5WebKit.so.5.212.0 /usr/lib64/libQt5Network.so.5.12.4 /usr/lib64/libQt5PrintSupport.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 src/corelibs/U2Core/libU2Core.so src/libs_3rdparty/sqlite3/libugenedb.so src/libs_3rdparty/zlib/libzlib.a /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Network.so.5.12.4 /usr/lib64/libQt5Script.so.5.12.4 /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 /usr/lib64/libQt5Widgets.so.5.12.4 /usr/lib64/libQt5Gui.so.5.12.4 /usr/lib64/libQt5Xml.so.5.12.4 /usr/lib64/libQt5Core.so.5.12.4 && :
/usr/bin/ld: src/ugenecl/CMakeFiles/ugenecl.dir/src/Main.cpp.o: in function `U2::DashboardInfoRegistry::DashboardInfoRegistry()':
Main.cpp:(.text._ZN2U221DashboardInfoRegistryC2Ev[_ZN2U221DashboardInfoRegistryC5Ev]+0x20): undefined reference to `vtable for U2::DashboardInfoRegistry'
collect2: greška: ld returned 1 exit status
ninja: build stopped: cannot make progress due to previous errors.
```

**Comment by atiunov** [comment]
I've committed some fixes, but there are still many problems, if you build UGENE with CMake. I've managed to build and run the application, but there were troubles with translations and plugins. I don't think you'll be able to use UGENE built with CMake.
Use the following command to build UGENE with qmake:
`qmake UGENE_EXCLUDE_LIST_ENABLED=1 CONFIG+=x64 && make release`
We'll try to fix all issues related to CMake later.
